apps/prairielearn/src/middlewares/enterpriseOnly.ts

```diff
@@ -0,0 +1,8 @@
+import { isEnterprise } from '../lib/license.js';
+
+export default function enterpriseOnlyMiddleware(load) {
+  if (isEnterprise()) {
+    return load();
+  }
+  return (req, res, next) => next();
+}
```

Comment:
Two things:

- Now that this lives in a TypeScript file, let's add types! If you import `type RequestHandler` from `express`, you can type `load` as `() => T` and the return type as `T | RequestHandler`. That will involve making the function [generic](https://www.typescriptlang.org/docs/handbook/2/generics.html). In fact, if you look at where this function came from in `server.js`, there were already JSDoc types there!
- Let's export this as a named function rather than a default one.

Reply:
Added these types in [15771e5](https://github.com/PrairieLearn/PrairieLearn/pull/10906/commits/15771e55e6dd9050a1d39214df8424edc073a18f). Also, changed this from a default function to a named one.